framework/src/bundle/CoreBundleContext.h

```diff
@@ -197,6 +233,13 @@ namespace cppmicroservices
         // The core context is exclusively constructed by the FrameworkFactory class
         friend class FrameworkFactory;
 
+        // Mutex required to be held when changing stopped.
+        // ReadLock or WriteLock construction is done using this mutex.
+        mutable std::shared_mutex stoppedLock;
+
+        // Flag for whether the Framework has been stopped. See mutex stoppedLock
+        bool stopped;
```

Comment:
I see that the bundle queries its own state. The crux here is the communication between any given `BundlePrivate` and the `FrameworkPrivate`. I think that the `BundlePrivate::state` variable is only owned and visible to that bundle and therefore is not going to be able to link with changes made in the `Frameworkprivate`. The `CoreBundleContext` is the variable that I found that linked those two independent objects. Let me know if this is not your understanding because yes it would be ideal to not have to introduce new methods etc here.